### ISSUE TITLE
fastrtpsgen: example generation copies consts from a msg to another that uses a builtin type [5433]

### ISSUE THREAD

**Issue by TSC21** [eProsima/Fast-RTPS#534]
Hi,

I have an IDL file with the following structure:
```
#ifndef __position_setpoint__idl__
#define __position_setpoint__idl__

const octet position_setpoint__SETPOINT_TYPE_POSITION = 0;
const octet position_setpoint__SETPOINT_TYPE_VELOCITY = 1;
const octet position_setpoint__SETPOINT_TYPE_LOITER = 2;
const octet position_setpoint__SETPOINT_TYPE_TAKEOFF = 3;
const octet position_setpoint__SETPOINT_TYPE_LAND = 4;
const octet position_setpoint__SETPOINT_TYPE_IDLE = 5;
const octet position_setpoint__SETPOINT_TYPE_OFFBOARD = 6;
const octet position_setpoint__SETPOINT_TYPE_FOLLOW_TARGET = 7;
const octet position_setpoint__VELOCITY_FRAME_LOCAL_NED = 1;
const octet position_setpoint__VELOCITY_FRAME_BODY_NED = 8;

struct position_setpoint_
{
    unsigned long long timestamp_;
    boolean valid_;
    octet type_;
    float x_;
    float y_;
    float z_;
    boolean position_valid_;
    float vx_;
    float vy_;
    float vz_;
    boolean velocity_valid_;
    octet velocity_frame_;
    boolean alt_valid_;
    double lat_;
    double lon_;
    float alt_;
    float yaw_;
    boolean yaw_valid_;
    float yawspeed_;
    boolean yawspeed_valid_;
    octet landing_gear_;
    float loiter_radius_;
    octet loiter_direction_;
    float pitch_min_;
    float a_x_;
    float a_y_;
    float a_z_;
    boolean acceleration_valid_;
    boolean acceleration_is_force_;
    float acceptance_radius_;
    float cruising_speed_;
    float cruising_throttle_;
}; // struct position_setpoint_

#pragma keylist position_setpoint_
#endif  // __position_setpoint__idl__
```

And another IDL file that imports the previous IDL file for its structure:
```
#ifndef __position_setpoint_triplet__idl__
#define __position_setpoint_triplet__idl__

#include "position_setpoint_.idl"

struct position_setpoint_triplet_
{
    unsigned long long timestamp_;
    position_setpoint_ previous_;
    position_setpoint_ current_;
    position_setpoint_ next_;
}; // struct position_setpoint_triplet_

#pragma keylist position_setpoint_triplet_
#endif  // __position_setpoint_triplet__idl__
```
When generating the example Fast-RTPS app for the second one using `fastrtpsgen -example x64Linux2.6gcc <relative_path_to_idl_dirs>/position_setpoint_triplet_.idl` (after of course generating the same for the `position_setpoint_.idl`), it comes to my attention that `position_setpoint_triplet_.h` has in its content the same constants as `position_setpoint_.h`, which in a build process where I add both to the same target, results on a redefinition build error as reported in https://github.com/PX4/Firmware/pull/12046#issuecomment-493776537.

This leads me to think there's so weird issue with the preprocessor in this case. I tried to use `-I` to setup preprocessor include dir but with the same result.

Is there a way of understanding about what's going on here? I expect that the second IDL, which has builtin type of another IDL file, would result on having the definitions of the first included on the second, but not repeated (in what concerns to the constants in this case).

Thanks in advance for this!

**Comment by TSC21** [comment]
@LuisGP any insights? Thank you!

**Comment by TSC21** [comment]
Btw I am using Fast-RTPS 1.7.2.

**Comment by LuisGP** [comment]
Hi @TSC21 
We are taking a look into this issue. We have been able to reproduce it. I hope we can release a patch soon.

Thank you for reporting the issue.
Best regards!

**Comment by TSC21** [comment]
> Hi @TSC21
> We are taking a look into this issue. We have been able to reproduce it. I hope we can release a patch soon.
> 
> Thank you for reporting the issue.
> Best regards!

Alright. Let me know if I can help. Thank you!

**Comment by LuisGP** [comment]
Hi @TSC21 

I just created the PR [#49](https://github.com/eProsima/Fast-CDR/pull/49) in Fast-CDR that should fix the issue.

I would appreciate if you can test it.

Best regards.

**Comment by TSC21** [comment]
@LuisGP just tested https://github.com/eProsima/Fast-RTPS/pull/537 and it seems to the job. Will this be an hotfix to the 1.8.0 release?

**Comment by TSC21** [comment]
@richiware I know the fix for this eas released in 1.8.1. But for the ROS2 use cases, if I want to connect with ROS2 Bouncy or Crystal, the versions won't match. So could this be an hotfix to be released for 1.6.0 and above as well? Or do you have another alternative for this? Thanks in advance.
